org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/settings/SettingsMergeTest.java

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.settings;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+
+import org.eclipse.lemminx.extensions.contentmodel.settings.XMLValidationSettings;
+import org.junit.jupiter.api.Test;
+
+public class SettingsMergeTest {
+
+	@Test
+	public void testValidationSettingsMerge() throws IllegalArgumentException, IllegalAccessException {
+		XMLValidationSettings settings = new XMLValidationSettings();
+		mutateValues(settings);
+		XMLValidationSettings mergeTargetSettings = new XMLValidationSettings();
+		mergeTargetSettings.merge(settings);
+		assertEquals(settings, mergeTargetSettings);
+	}
+
+	/**
+	 * Mutates all the values of all the fields, under the assumption that all Object fields are non-null
+	 */
+	public static void mutateValues(Object obj) throws IllegalArgumentException, IllegalAccessException {
+		for (Field field : XMLValidationSettings.class.getFields()) {
+			Type type = field.getGenericType();
+			switch (type.getTypeName()) {
+			case "boolean":
+				field.set(obj, !(Boolean)field.get(obj));
+				break;
+			case "int":
+			case "short":
+			case "long":
+			case "char":
+			case "byte":
+			case "float":
+			case "double":
+				field.set(field, (Integer)field.get(obj) == 0 ? 1 : 0);
+				break;
+			default:
```

Comment:
noGrammar being a String would probably fall under this case, but because it's already null, I don't think it would catch the case of it not being properly set.